extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+			$leftColumn.sortable(options);
+			options['connectWith'] = '.lcs-container';
+			$rightColumn.sortable(options);
+			$leftColumn.sortable('enable');
+			$rightColumn.sortable('enable');
+			$momBarContent.each(function () {
+				$(this).attr('contenteditable', true);
+			});
+		}, onSave = function () {
+			removeEmpty();
+			$editMode.hide();
+			$nonEditMode.show();
+			$mainContent.startThrobbing();
+			$moms.removeClass('mom-hidden');
+			$moms.removeClass('mom-edit');
+			$momBar.css('display', 'none');
+			$momOverlays.hide();
+			$leftColumn.sortable('disable');
+			$rightColumn.sortable('disable');
+			$momBarContent.each(function () {
+				$(this).removeAttr('contenteditable');
+			});
+			$('.mom-edit-btn').show();
+			$('.mom-delete-btn').show();
+			$('.mom-content').show();
+			$('.mom-save-btn').hide();
+			$('.mom-discard-btn').hide();
+			$('.mom-wiki-markup').hide();
+			var left = [],
+				right = [];
+			$('.lcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					left.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar .mom-bar-content').text()
+					}));
+				}
+			});
+			$('.rcs-container .mom-module').each(function () {
+				var data = $(this).data();
+				if (typeof data.title != 'undefined') {
+					right.push(JSON.stringify({
+						'title': data.title,
+						'text': $(this).find('.mom-bar-content').text()
+					}));
+				}
+			});
+			$.nirvana.sendRequest({
+				controller: 'NjordController',
+				method: 'reorder',
+				type: 'POST',
+				data: {
+					'page': wgTitle,
+					'left': left,
+					'right': right
+				},
+				callback: onDataSaved,
+				onErrorCallback: function () {
+					// TODO: handle failure
+				}
+			});
+		}, onDataSaved = function () {
+			$mainContent.stopThrobbing();
+		}, onScroll = function () {
+			if ($(window).scrollTop() >= $mainContent.offset().top) {
+				$momHeader.addClass('mom-fixed');
+			} else {
+				$momHeader.removeClass('mom-fixed');
+			}
+		}, onDiscard = function () {
+			$editMode.hide();
+			$nonEditMode.show();
+			$mwContent.replaceWith($clonedMain);
+			$mwContent.refresh();
+			$mainContent.refresh();
+			$deleteButton.refresh();
+			$momOverlays.refresh();
+			$momBar.refresh();
+			$momBarContent.refresh();
+			$moms.refresh();
+			$leftColumn.refresh();
+			$rightColumn.refresh();
+			$deleteButton.on('click', function () {
+				$(this).parents('.mom-module').remove();
+			});
+			$('.mom-save-btn').hide();
+			$('.mom-discard-btn').hide();
+			$('.mom-wiki-markup').hide();
+		}, addEmpty = function () {
+			var $new = $(document.createElement('div')),
+				$button = $(document.createElement('div'));
+			$button.text('ADD');
+			$button.addClass('add-btn new-btn');
+			$new.addClass('mom-module mom-add-module');
+			var $secButton = $button.clone(true),
+				$secNew = $new.clone();
+			$button.on('click', {align: 'right'}, addNewModule);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "addNewModule" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)